### PR TITLE
Add :owner to DBConnection.Connection, closes #1

### LINKED ISSUE #1: Add owner feature to connection
See: https://github.com/robconery/moebius/issues/37

If the owner dies, the connection process should exit.

### PR DESCRIPTION
**Pull request by josevalim**

### CODE REVIEW

File: lib/db_connection/connection.ex [hunk 1]
```diff
@@ -93,6 +93,11 @@ defmodule DBConnection.Connection do
   def init({mod, opts, mode, info}) do
     queue  = if mode == :sojourn, do: :broker, else: :queue.new()
     broker = if mode == :sojourn, do: info
+
+    if owner = Keyword.get(opts, :owner) do
+      Process.link(owner)
```
Comment (fishcakez) [inline lib/db_connection/connection.ex]: Can we monitor instead and store the ref in state? Can't rule callback module trapping exists.

Comment (josevalim) [inline lib/db_connection/connection.ex]: Ah yes, forgot about it. In case of crashes, should we just exit with the same reason as in the DOWN message?

Comment (fishcakez) [inline lib/db_connection/connection.ex]: Yes lets do that.

Comment (fishcakez) [inline lib/db_connection/connection.ex]: Or just stop with `:shutdown` might be fine here.

**Comment by josevalim** [comment]
Updated.

File: lib/db_connection/connection.ex [hunk 2]
```diff
@@ -93,8 +93,12 @@ defmodule DBConnection.Connection do
   def init({mod, opts, mode, info}) do
     queue  = if mode == :sojourn, do: :broker, else: :queue.new()
     broker = if mode == :sojourn, do: info
+
+    owner     = Keyword.get(opts, :owner)
+    owner_ref = owner && Process.monitor(owner)
```
Comment (fishcakez) [inline lib/db_connection/connection.ex]: we should check `mode == :connection` so poolled workers dont get an owner

**Comment by josevalim** [comment]
Updated.

**Pull request merged**